OpenSim/Common/Component.h

```diff
@@ -935,6 +947,17 @@ template <class T> friend class ComponentMeasure;
         constructOutputs();
     }
 
+    /**
+    * Mark another Component as a subcomponent of this Component. Component
+    * methods (e.g. addToSystem(), initStateFromProperties(), ...) are
+    * therefore invoked on subcomponents when called on this Component.
+    * Realization is also performed automatically on subcomponents. This
+    * member function does not take ownership, it is assumed that the
+    * component is a data member or property of this Component.
+    */
+    void markAsSubcomponent(Component* component);
```

Comment:
I don't understand why this is a pointer if there is no ownership change happening. Shouldn't it be a reference? Also in view of the PR description, this seems wrong since it would allow "marking" of a non-owned subcomponent. I'm not sure I get the term "mark" here either -- my understanding is that any component owned by a parent is a subcomponent of that parent. If that's right then shouldn't subcomponent-ness be a side effect of ownership?

Is this just an obsolete method at this point?